### PR TITLE
Poseidon2 Hash

### PR DESCRIPTION
**Pull request by vivekvpandya**

### CODE REVIEW

**Comment by vivekvpandya** [comment]
Just to note , on comparing generated asm code against horizen implementation I see https://discourse.llvm.org/t/rfc-cmov-vs-branch-optimization/6040 kind of optimization opportunity in plonky2 generated posiden2 code.

File: plonky2/benches/poseidon_prove.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+                let data = builder.build::<C>();
+
+                let _proof = data.prove(pw);
+            });
+        });
+    }
+}
+
+pub fn bench_poseidon_remove_prove<
```
Comment (sai-deng) [inline plonky2/benches/poseidon_prove.rs:67]: What is the difference of this function?

**Review by sai-deng — APPROVED** [review]
Could you please acknowledge the source of the copied code within the files? It would be helpful to include comments indicating that the code was derived from Ola VM, along with a link to the original repository. This way, we maintain transparency and provide a clear reference to the original work. Thank you!

File: plonky2/benches/poseidon_prove.rs (hunk 1 above)
Comment (vivekvpandya) [reply plonky2/benches/poseidon_prove.rs:67]: It just skips prove() call. so only benches generation.

**Pull request merged**